Lib/textwrap.py

```diff
@@ -402,10 +417,43 @@ def shorten(text, width, **kwargs):
         >>> textwrap.shorten("Hello  world!", width=11)
         'Hello [...]'
     """
-    w = TextWrapper(width=width, max_lines=1, **kwargs)
+    w = TextWrapper(width=width, cjk=cjk, max_lines=1, **kwargs)
     return w.fill(' '.join(text.strip().split()))
 
 
+# -- CJK support ------------------------------------------------------
+
+def cjk_wide(char):
+    """Return True if char is Fullwidth or Wide, False otherwise.
+    Fullwidth and Wide CJK chars are double-width.
+    """
+    import unicodedata
+    return unicodedata.east_asian_width(char) in ('F', 'W')
```

Comment:
Please use some other type than a tuple (e.g. persistent frozenset should be faster here -> you can just use `{'F', 'W'}`)